prime-router/src/main/kotlin/azure/EmailEngineFunction.kt

```diff
@@ -136,16 +137,36 @@ class EmailScheduleEngine {
 
             // get the orgs to fire for
             orgs.forEach { org ->
+                val countOfRecords = workflowEngine.db.fetchDownloadableReportFiles(
+                    OffsetDateTime.now().minusDays(1L),
+                    org
+                ).size;     
+                val template = getTemplate(schedule.template, countOfRecords );     
                 val emails: Iterable<String> = if (schedule.emails.size > 0) schedule.emails else getEmails(org, logger)
-                logger.info("EmailEngineFunction:: processing organization $org within ${schedule.template}")
+                logger.info("EmailEngineFunction:: processing organization $org within ${template}")
                 emails.forEach { email ->
-                    logger.info("EmailEngineFunction:: sending email to $email")
-                    dispatchToSendGrid(schedule.template, listOf(email), logger)
+                    logger.info("EmailEngineFunction:: sending email template ${template} to $email")
+                    dispatchToSendGrid(template, listOf(email), logger)
```

Comment:
`dispatchToSendGrid` accepts any `Iterable<String>`, why the explicit conversion to a list?

cf. https://github.com/CDCgov/prime-reportstream/blob/b15e028ff7a9cfa8b2fa023a6c119251eda734a8/prime-router/src/main/kotlin/azure/EmailEngineFunction.kt#L327
```kotlin
    private fun dispatchToSendGrid(
        template: String,
        emails: Iterable<String>,
        logger: Logger
    )
```

Reply:
Iterable is an interface - need a tangible instance

Reply:
ah, apologies, I thought `email` was already iterable, but it's just a `string`, and so you're turning the one string into a list of strings.